library/patient_self_checkin/step5.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+use OpenEMR\Core\Header;
+
+$ignoreAuth = true;
+require_once "../../interface/globals.php";
+if (!$GLOBALS['self_checkin_enable'] == 1) {
+    die("This feature has not been enabled");
+}
+?>
+
+<link rel="stylesheet" type="text/css" href="/interface/themes/selfCheckIn.css">
+
+<!DOCTYPE html>
+<html>
+<head>
+
+    <title><?php echo xlt("Check In"); ?></title>
+<?php
+
+Header::setupHeader();
+
+?>
+</head>
+<body class="step5">
+
+<p>
+    <img style="text-align: center"
+         width="500px" src='<?php echo $GLOBALS['images_static_relative']; ?>/logo-full-con.png'/>
+</p>
+
+<?php
+
+$month = htmlspecialchars($_GET['month']);
+$birthday = htmlspecialchars($_GET['date']);
+$surname = htmlspecialchars($_GET['surname']);
```

Comment:
Above 3 lines are a good example of why it's best to do the html escaping only:
1. When output the variable to the html
2. Or earlier if building a string that contains html elements in it

Otherwise, there is risk of having weird bugs if run html escaped code through coding algorithms, such as running them through the sql queries below.

Reply:
OpenEMR also has 2 shortcut html escaping functions, which will cover below.